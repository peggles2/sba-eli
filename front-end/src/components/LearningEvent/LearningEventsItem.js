import React, { Component } from "react";

export default class LearningEventsItem extends Component {
  render() {
    return <li>Learning Event: {this.props.item.title}</li>;
  }
}
