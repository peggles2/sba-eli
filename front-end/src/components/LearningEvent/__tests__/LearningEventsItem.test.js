import React from "react";
import LearningEventsItem from "../LearningEventsItem";
import { shallow } from "enzyme";

describe("LearningEventsItem", () => {
  it("should render correctly", () => {
    const item = { id: 1, title: "LE 1" };
    const wrapper = shallow(<LearningEventsItem item={item} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should render a <li>", () => {
    const item = { id: 1, title: "LE 1" };
    const wrapper = shallow(<LearningEventsItem item={item} />);

    expect(wrapper.find("li").length).toEqual(1);
  });
});
