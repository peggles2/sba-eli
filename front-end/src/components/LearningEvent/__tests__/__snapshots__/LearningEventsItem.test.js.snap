// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningEventsItem should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningEventsItem
    item={
      Object {
        "id": 1,
        "title": "LE 1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Learning Event: ",
        "LE 1",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Learning Event: ",
      "LE 1",
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Learning Event: ",
          "LE 1",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Learning Event: ",
        "LE 1",
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
